### ISSUE TITLE
Add support for GPG Signing

### ISSUE THREAD

**Issue by mrchief** [googleapis/release-please#1314]
Bubbling up this issue upstream https://github.com/google-github-actions/release-please-action/issues/399

**Comment by IchordeDionysos** [comment]
@mrchief @chingor13 this should be unblocked now, correct?
One can supply a commit signer.

**Comment by justinthelaw** [comment]
Any updates on this?

**Comment by IchordeDionysos** [comment]
@justinthelaw I have built an implementation for commit signing two weeks ago.

I'm waiting for a review regarding the implementation and the general API interface


/cc @chingor13

**Comment by thomasnemer** [comment]
Hi there :wave: 
Would that also work with SSH commit signature? As I understand, `release-please` uses `code-suggester` to create the commit. I've seen [something](https://github.com/googleapis/code-suggester/blob/f091b69577a0ad2d14eb90888485f28909aba480/src/github/create-commit.ts#L48) on `code-suggester`'s side about signature, it seems to use github's REST api for commits, but I can't figure out if it's compatible with SSH signature (even if Github is)

**Comment by IchordeDionysos** [comment]
@thomasnemer Yes, correct `release-please` uses `code-suggester.`

However, the function you linked with the `signer.generateSignature()` does not pass all the necessary information to sign commits. It is missing the crucial date of the commit which is required to sign it.
There is an open PR to alter `code-suggester` to expose this: https://github.com/googleapis/code-suggester/pull/485

Once that's finished/merged, we can think about signing the commits.
I have an example implementation using GPG keys here: https://github.com/googleapis/code-suggester/issues/337#issuecomment-1917754635

Something similar could probably be built for SSH or S/MIME.
For SSH, this could be a helpful library: https://www.npmjs.com/package/ssh-signer (haven't tested it)

-- 

So, TL;DR:
- Yes it should be possible to sign commits using SSH
- We need to merge: 
  https://github.com/googleapis/code-suggester/pull/485 to enable signing at all
- We need to build two implementations for signing using GPG and SSH
- Expose those implementations somehow in release-please configurations

**Comment by justinthelaw** [comment]
@thomasnemer have you already requested approval from one of the google-apis maintainers to kick-off the checks and review your PR? Seems like those are the 2 blockers: maintainer pipeline kick-off and maintainer review.

**Comment by justinthelaw** [comment]
Apologies, I meant to @IchordeDionysos not you @thomasnemer haha

**Comment by IchordeDionysos** [comment]
@justinthelaw I have mentioned @chingor13 in this comment:
https://github.com/googleapis/code-suggester/issues/337#issuecomment-1917754635 who seems to be the primary contributor
